src/stan/mcmc/chains.hpp

```diff
@@ -380,10 +378,12 @@ class chains {
       throw std::invalid_argument(
           "add(stan_csv): number of columns in"
           " sample does not match chains");
-    if (!param_names_.cwiseEqual(stan_csv.header).all()) {
-      throw std::invalid_argument(
-          "add(stan_csv): header does not match"
-          " chain's header");
+    for (int i = 0; i < num_params(); i++) {
+      if (param_names_[i] != stan_csv.header(i)) {
+        throw std::invalid_argument(
```

Comment:
[optional]
This is only optional because it's feature creep on the actual issue being fixed.  

We should build up a message including the parameter names that don't match and the index `i`.  That can be done with a stringstream.  It's acceptable in terms of performance to format a message within the body of the conditional where an exception will be thrown.

Reply:
I went ahead and changed the error message while I was touching this code.